app/src/rgb_underglow.c

```diff
@@ -468,17 +468,28 @@ int zmk_rgb_underglow_change_spd(int direction) {
 
 #if IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_AUTO_OFF_IDLE) ||                                          \
     IS_ENABLED(CONFIG_ZMK_RGB_UNDERGLOW_AUTO_OFF_USB)
-static int rgb_underglow_auto_state(bool *prev_state, bool new_state) {
-    if (state.on == new_state) {
+struct rgb_underglow_sleep_state {
+    bool sleeping;
+    bool pre_sleep;
+};
+
+static int rgb_underglow_auto_state(bool is_wakeup_event) {
+    static struct rgb_underglow_sleep_state sleep_state = {sleeping : false, pre_sleep : false};
+
+    // wake up event while awake, or sleep event while sleeping -> no-op
+    if (!is_wakeup_event == sleep_state.sleeping) {
```

Comment:
To be honest, the naming of `is_wakeup_event` and then using `sleeping` as the state leads to this line which is really counterintuitive when reading quickly. I would much rather either:

`bool  target_wake_state` and then store a `bool awake` in the structure, so that we are comparing the same logical  state, and don't need negation as found here.

Thoughts?